### ISSUE TITLE
bug: sheet modal gesture breaks when interacting with inputs/keyboard on ios

### ISSUE THREAD

**Issue by EinfachHans** [ionic-team/ionic-framework#23878]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

I just tested the new Sheet Modal. First: Great work 🎉 Here is a Collections of things i mentioned:
- Similar to #23561: My IDEA (InteliJ) seems to not know the new properties like `breakpoints` or `initialBreakpoint`. Maybe missing in the angular component declaration?
- The `handle` can not be set to **false** for Sheet Modals: https://github.com/ionic-team/ionic-framework/blob/9fa373c6efeceb39913f1c158cda383c38cba29c/core/src/components/modal/modal.tsx#L550
- The handle itself doesn't looks nice in combination with IonTitle - Looks like it needs a bit more spacing here <img src="https://user-images.githubusercontent.com/20899975/132209662-0ba88661-3bfc-49fb-a2b0-2c31e3f0a52c.png" width="200">
- The Backdrop seems to be a bit buggy yet, see video: After drag finished it changes:
<img src="https://user-images.githubusercontent.com/20899975/132210708-3a249757-20d1-4507-be9a-e663ccafc871.mov" width="200">

- When i have an Input inside the Sheet Modal and focus it the Keyboard appears and the Modal pushs up a bit. After losing the focus of the input and the keyboard is gone, the breakpoints do not work anymore and i'm kind of in a free mode where i can stop the Modal wherever i want
- Where we already talk about that: This `freeMode` would be a cool feature
- Also it would be great to be able to disable the possibility to drag the Modal and add the possibility to change the current breakpoint programatically.

### Expected Behavior

Everything is described above

### Steps to Reproduce

Version: `6.0.0-beta.5`

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you please provide reproductions for the buggy backdrop and the keyboard issues?

**Comment by EinfachHans** [comment]
Of course. Here is the Repo: https://github.com/EinfachHans/sheet-modal-bugs

Just open the Modal, and move it in the states, the backdrop bug happens. Then tap into the input and tap out again. The `freeMode` is then enabled 🤪 (See [Video](https://user-images.githubusercontent.com/20899975/132634178-6a1a6fb3-7362-464f-9e83-3358e6c8a588.mp4))

Tested on Android 11 Galaxy 4 and iOS 14.7.1 (iPhone 11 Pro):
- Backdrop Bug on both
- FreeMode Bug only on iOS

**Comment by liamdebeasi** [comment]
Thanks. Looks like the backdrop issue happens when the breakpoints do not range from 0 to 1 (you had `[0, 0.4, 0.7]`. I need to check with the team to see if this is something we want to support. Do you have a use case for why you would want part of the modal hidden off screen?


I can reproduce the issue with the sheet not snapping, though I need to look into why that is happening.

**Comment by EinfachHans** [comment]
I would like to use the Sheet Modal for example single options, where the modal itself only contains one button. This should only have a specific height of for example **0.5**, because expanding to **11** won't look really good.

**Comment by liamdebeasi** [comment]
I think it would be better to set the height of the modal to `50%` rather than have the height be `100%` and have half of the modal hidden off screen:

```html
<ion-modal [breakpoints]="[0, 1]">...</ion-modal>
```

```css
ion-modal {
  height: 50%;
}
```

The breakpoints for `ion-modal` are relative to the height of the modal so that you do not need to re-adjust the breakpoints every time you change the height of the modal. (The docs for this are wrong, but that will be fixed next beta)

Can you try adjusting the height and let me know if that works?

**Comment by liamdebeasi** [comment]
Looks like the same issue as https://github.com/ionic-team/ionic-framework/issues/22120 (impacts the card modal too)

**Comment by EinfachHans** [comment]
Yes correct!

**Comment by liamdebeasi** [comment]
Some updates on this:

1. IDE issue is resolved via https://github.com/ionic-team/ionic-framework/pull/23899
2. Handle position is resolved via https://github.com/ionic-team/ionic-framework/pull/23901
3. Disabling handle is resolved via https://github.com/ionic-team/ionic-framework/pull/23900
4. Backdrop animation when breakpoints does not include `1` is resolved via https://github.com/ionic-team/ionic-framework/pull/23927

The above 4 fixes will be available in an upcoming Ionic v6 beta.

The "bounce" you mentioned in https://github.com/ionic-team/ionic-framework/issues/23878#issuecomment-916126140 will be tracked in https://github.com/ionic-team/ionic-framework/issues/22120 since it also impacts card modals.
Programmatically moving the breakpoint is currently being discussed in another thread you have (https://github.com/ionic-team/ionic-framework/issues/23917).

I am going to convert this thread to focus on the remaining issue which is the sheet gesture breaking when Interacting with the keyboard/inputs.

**Comment by EinfachHans** [comment]
@liamdebeasi why is this removed from the 6.0.0 Milestone? Guess thats a major issue 🤔

**Comment by liamdebeasi** [comment]
We still have plans to fix, but the 6.0.0 milestone was outdated and not representative of what we are currently working on.

**Comment by EinfachHans** [comment]
@liamdebeasi FYI, looks like the `canStart` of the sheet gesture is called when an input inside of the sheet style modal is clicked and the keyboard opens 🤔

Additional to that, i guess it would make sense to close open keyboards when start dragging by adjusting the canStart:

```ts
const canStart = (detail: GestureDetail) => {
  /**
   * If the sheet is fully expanded and
   * the user is swiping on the content,
   * the gesture should not start to
   * allow for scrolling on the content.
   */
  const content = (detail.event.target! as HTMLElement).closest('ion-content');

  if (currentBreakpoint === 1 && content) {
    return false;
  }

  const activeElement = baseEl.ownerDocument!.activeElement as HTMLElement;
  if (activeElement && activeElement.matches('input,ion-input, ion-textarea')) {
    activeElement.blur();
  }

  return true;
};
```
